### PR TITLE
upgrade modules with known vulnerabilities

### PR DESCRIPTION
**Pull request by krancour**
This PR updates several development-time dependencies and one runtime dependency to remediate known vulnerabilities.

There is one vulnerability that cannot be remediated at this time. I've split the unit tests and the audit into separate jobs so it's clear that the audit is failing and that unit tests are passing.

### CODE REVIEW

File: brigade.js [hunk 1]
```diff
@@ -2,7 +2,7 @@ const { events, Job } = require("@brigadecore/brigadier")
 const { Check } = require("@brigadecore/brigade-utils");
 
 const projectName = "brigadier"
-const nodeImage = "node:12.3.1-stretch"
+const nodeImage = "node:16.13.0-bullseye"
```
Comment (krancour) [inline brigade.js]: Upgrading `typedoc` required a newer version of node-- at least 12.10.0. I made a judgement call that there isn't a lot of harm in upgrading to the latest stable version.

File: brigade.js [hunk 2]
```diff
@@ -12,16 +12,43 @@ function build() {
...
         "yarn install",
         "yarn compile",
-        "yarn test",
+        "yarn test"
+    ];
+
+    return build;
+}
+
+function audit() {
```
Comment (krancour) [inline brigade.js:21]: There is one moderate vulnerability that cannot be remediated at this time.

I figured as long as that's the case, we should separate the audit and the unit tests from one another. This way, it's more clear, at a glance, that tests are passing. fwiw, we did this a long time ago on Brigade itself.

File: brigade.js [hunk 3]
```diff
@@ -12,16 +12,43 @@ function build() {
...
+    // Note: as provided language string is used in job naming, it must consist
+    // of lowercase letters and hyphens only (per Brigade/K8s restrictions)
+    return Promise.all([
+        new Check(e, p, build()).run().catch((err) => { return err }),
+        new Check(e, p, audit()).run().catch((err) => { return err })
+    ]).then((values) => {
+        values.forEach((value) => {
+            if (value instanceof Error) throw value;
+        });
+    });
```
Comment (krancour) [inline brigade.js:51]: See https://github.com/brigadecore/brigadier/pull/54/files#r758615701

This is copy/paste from Brigade itself and follows a pattern we established a long time ago.

File: package.json [hunk 4]
```diff
@@ -24,19 +24,19 @@
...
     "rimraf": "^2.6.2",
     "ts-node": "^7.0.0",
-    "typedoc": "^0.15.0",
-    "typescript": "^3.2.2"
+    "typedoc": "^0.22.10",
+    "typescript": "^4.5.2"
   },
   "dependencies": {
-    "@kubernetes/client-node": "^0.10.1",
+    "@kubernetes/client-node": "^0.16.1",
```
Comment (krancour) [inline package.json]: Everything upgraded here had vulnerabilities in transitive dependencies.

File: src/logger.ts [hunk 5]
```diff
@@ -1,4 +1,4 @@
-import * as PrettyError from "pretty-error";
+import PrettyError from "pretty-error";
```
Comment (krancour) [inline src/logger.ts]: Had to do this after upgrading `@kubernetes/client-node`.

File: tsconfig.json [hunk 6]
```diff
@@ -2,6 +2,7 @@
     "compilerOptions": {
         "module": "commonjs",
         "target": "es6",
+        "esModuleInterop": true,
```
Comment (krancour) [inline tsconfig.json]: See https://github.com/brigadecore/brigadier/pull/54/files#r758617493

**Review by vdice — APPROVED** [review]

File: brigade.js (hunk 1 above)
Comment (krancour) [reply brigade.js]: Reconsidering this, because there _have_ been breaking changes in TypeScript.

Comment (krancour) [reply brigade.js]: err... was momentarily confusing Node and TypeScript.

I'm still, after more playing, finding it's better to make the most minimal changes possible here, so I'm only going to bump this to the latest 12.x.y.

File: src/logger.ts (hunk 5 above)
Comment (krancour) [reply src/logger.ts]: But upgrading `@kubernetes/client-node` doesn't improve the situation vis-a-vis its transitive dependency that has some issues, so I'm going to not both with that and then this change also isn't needed.

**Pull request merged**